services/shhext/api.go

```diff
@@ -441,3 +445,30 @@ func makePayload(r MessagesRequest) []byte {
 
 	return append(data, cursorBytes...)
 }
+
+func createBloomFilter(r MessagesRequest) []byte {
+	if len(r.Topics) > 0 {
+		return topicsToBloom(r.Topics...)
+	}
+
+	return whisper.TopicToBloom(r.Topic)
+}
+
+func topicsToBloom(topics ...whisper.TopicType) []byte {
+	combined := make([]byte, whisper.BloomFilterSize)
+
+	for _, t := range topics {
+		bloom := whisper.TopicToBloom(t)
+
+		i := new(big.Int).SetBytes(combined)
+		i.Or(i, new(big.Int).SetBytes(bloom[:]))
+
+		data := i.Bytes()
+		newCombined := make([]byte, whisper.BloomFilterSize)
```

Comment:
is there a point of creating this slice? it seems work can be done on `combined` directly